lib/ansible/plugins/action/set_fact.py

```diff
@@ -45,10 +46,11 @@ def run(self, tmp=None, task_vars=None):
             for (k, v) in iteritems(self._task.args):
                 k = self._templar.template(k)
 
-                if not isidentifier(k):
+                try:
```

Comment:
@bcoca I put the `validate_variable_names` check in `VariableManager.set_nonpersistent_facts` which `set_fact` uses in the results processing in the strategy. I guess we should remove it from here so we don't double validate the varname (and double warn on a reserved name) but that would prevent users from doing `ignore_errors` on `set_fact` for this particular error, which is a regression. What do you think?

Reply:
double warn will not happen (display.warning uniques) but i would try to avoid double processing if possible, i would put both checks in set_nonpersistent_facts one and remove them from action plugins